src/app/views/datasets-new/edit-metadata-cantabular/CantabularMetadataController.jsx

```diff
@@ -0,0 +1,755 @@
+                related_datasets: cantResponse.table_query_result.service.tables[0].meta.related_datasets,
+                publications: cantResponse.table_query_result.service.tables[0].meta.publications,
+                unit_of_measure: cantResponse.table_query_result.service.tables[0].meta.statistical_unit.statistical_unit,
+                release_frequency: cantResponse.table_query_result.service.tables[0].meta.release_frequency,
+                qmi: {
+                    href: cantResponse.dataset_query_result.dataset.meta.source.methodology_link,
+                },
+                contacts: [
+                    {
+                        name: cantResponse.dataset_query_result.dataset.meta.source.contact.contact_name,
+                        email: cantResponse.dataset_query_result.dataset.meta.source.contact.contact_email,
+                        telephone: cantResponse.dataset_query_result.dataset.meta.source.contact.contact_phone,
+                        website: cantResponse.dataset_query_result.dataset.meta.source.contact.contact_website,
+                    },
+                ],
+            },
+            version: {
+                dimensions: cantResponse.dataset_query_result.dataset.variables.edges.map(dimension => {
+                    return {
+                        name: dimension.node.name,
+                        description: dimension.node.meta.ons_variable.variable_description,
+                        label: dimension.node.name,
+                    };
+                }),
+                release_date: [year, month, day].join("/"),
+            },
+        };
+    };
+
+    handleGETSuccess = (nonCantDatasetMetadata, cantabularMetadata) => {
+        const mapped = this.mapMetadataToState(nonCantDatasetMetadata, cantabularMetadata);
+        if (mapped.state === "associated" && mapped.collection !== this.props.params.collectionID) {
+            this.setState({ disableScreen: true });
+            notifications.add({
+                type: "neutral",
+                message: `This dataset is in another collection.`,
+                isDismissable: true,
+            });
+        }
+        mapped.state === "created"
+            ? this.setState({ allowPreview: false, disableCancel: false })
+            : this.setState({ allowPreview: true, disableCancel: true });
+        this.setState({
+            metadata: mapped.metadata,
+            datasetIsInCollection: mapped.collection,
+            datasetCollectionState: mapped.datasetCollectionState,
+            versionCollectionState: mapped.versionCollectionState,
+            datasetState: mapped.state,
+            lastEditedBy: mapped.lastEditedBy,
+            versionIsPublished: mapped.versionIsPublished,
+        });
+    };
+
+    mapRelatedContentToState = (relatedDatasets, datasetID) => {
+        try {
+            return relatedDatasets.map((link, index) => {
+                return {
```

Comment:
There is a shorthand for returning an object that's useful when that's all a function has to do. So instead of

```
map((foo, bar) => {
  return {
    foo,
    bar
  };
})
```

you can wrap the object in parentheses and omit the `return` which is now implicit, like this:

```
map((foo, bar) => ({
  foo,
  bar
}))
```

Reply:
Done :white_check_mark: